src/gui/pedalboard/cables/CableView.h

```diff
@@ -51,6 +53,36 @@ class CableView : public Component,
     std::unique_ptr<CableViewPortLocationHelper> portLocationHelper;
 
     EditorPort nearestPort {};
+    juce::Point<int> portToPaint;
+
+    bool mouseOverClickablePort();
+    bool mouseDraggingOverOutputPort();
+
+    CriticalSection cableMutex;
+
+    struct PathGeneratorTask : juce::TimeSliceClient
+    {
+    public:
+        explicit PathGeneratorTask (CableView& cv) : cableView (cv)
+        {
+            sharedTimeSliceThread->addTimeSliceClient (this);
+
+            if (! sharedTimeSliceThread->isThreadRunning())
+                sharedTimeSliceThread->startThread();
+        }
+
+        int useTimeSlice() override;
+
+    private:
+        struct TimeSliceThread : juce::TimeSliceThread
+        {
+            TimeSliceThread() : juce::TimeSliceThread ("Cable Drawing Background Thread") {}
+        };
+
+        juce::SharedResourcePointer<TimeSliceThread> sharedTimeSliceThread;
```

Comment:
I wonder if we should make a more global `SharedTimeSliceThread` and have it be used by the cable view as well as the visualizations in the tuner and oscilloscope modules... anyway, we can figure that out later.

Reply:
Mmm yeah that would make sense good idea.